src/vl-tabs.js

```diff
@@ -179,24 +205,38 @@ export class VlTabs extends vlElement(HTMLElement) {
 
   __observeTabPanes(callback) {
     const observer = new MutationObserver(callback);
-    observer.observe(this, {childList: true});
+    observer.observe(this, {childList: true, subtree: true});
     return observer;
   }
 
   __processTabPane(mutations) {
     const tabPanesToAdd = mutations.flatMap((mutation) => [...mutation.addedNodes]).filter((node) => node instanceof VlTabsPane);
-    tabPanesToAdd.forEach((tabPane) => {
-      const index = this.__tabPanes.indexOf(tabPane);
-      this._addTab({id: tabPane.id, title: tabPane.title, index: index});
-      this._addTabSection({id: tabPane.id, index: index});
-    });
+    tabPanesToAdd.forEach((tabPane) => this.__addTabAndSection(tabPane));
+
     const tabPanesToDelete = mutations.flatMap((mutation) => [...mutation.removedNodes]).filter((node) => node instanceof VlTabsPane);
-    tabPanesToDelete.forEach((tabPane) => {
-      this._removeTab(tabPane.id);
-      this._removeTabSection(tabPane.id);
+    tabPanesToDelete.forEach((tabPane) => this.__removeTabAndSection(tabPane));
+
+    const tabPanesToUpdate = mutations.flatMap((mutation) => [...mutation.addedNodes])
+        .filter((node) => node.parentNode instanceof VlTabsPane)
+        .map((node) => node.parentNode);
+    tabPanesToUpdate.forEach((tabPane) => {
+      this.__removeTabAndSection(tabPane);
+      this.__addTabAndSection(tabPane);
     });
```

Comment:
Deze code lijkt overbodig?

Reply:
deze is in functie van 'test/unit/vl-tabs.test.html', 'tabs pane title slot wijzigingen updaten de tab' toegevoegd. Nu ik zag net het commentje van dit stukje code de test niet deed falen. Ik heb dit ondertussen aangepast. Te bekijken dus of die functionaliteit nodig is.

Reply:
Ik kan nog steeds deze code verwijderen zonder dat er unit of e2e testen falen, dus lijkt de code overbodig te zijn of er ontbreken vermoedelijk nog testen om aan te tonen waarvoor deze code nodig is?

Reply:
dit stukje wordt gebruikt om de title dynamisch te updaten. De code loopt hier door als ik de vl-tabs unit testen draai. Ik zal nog een test case aanmaken.

Reply:
ik zie nu net dat hier wel een unit test voor bestaat met naam 'tabs pane title slot wijzigingen updaten de tab' uit vl-tabs.test.html. Als ik dan weer de code in comment zet faalt die test weldegelijk.

![afbeelding](https://user-images.githubusercontent.com/1375279/118462743-4d9d4400-b6ff-11eb-80e5-e84f24c97b6f.png)

Firefox:
<img width="1552" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/118462789-5726ac00-b6ff-11eb-8307-18a6e75b7ebe.png">

Chrome:
<img width="1312" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/118462880-6efe3000-b6ff-11eb-8b47-49666b36de55.png">

Reply:
Als ik de code dan weer uncomment: 

![afbeelding](https://user-images.githubusercontent.com/1375279/118463035-98b75700-b6ff-11eb-85af-5587c46cfed0.png)

Wordt die test groen. 
Firefox:
<img width="1552" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/118463095-a836a000-b6ff-11eb-8ecb-84585e1edd8a.png">

Chrome: 
<img width="1312" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/118463177-c69c9b80-b6ff-11eb-8e21-a93768777c86.png">

Reply:
heb ook het gevoel dat we dit al besproken hadden tijdends de laatste conf call enige tijd geleden. Nu ja wat doe ik hier nu weer mee. Nog een testje schrijven, de demo uitbreiden, ...

Reply:
Ik heb vermoedelijk lokaal niet lang genoeg gewacht, want ik had voor de zekerheid de code in commentaar gecommit. Bamboo toont aan dat er wel een unit test faalt, dus de code is inderdaad nodig en blijkt ook uit de unit testen.

Reply:
Ik heb nog een code refactoring uitgevoerd.